sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/EqualPartitionsBalancingStrategy.java

```diff
@@ -55,30 +53,26 @@ public EqualPartitionsBalancingStrategy(String hostName, int minPartitionCount,
 
         int target = this.calculateTargetPartitionCount(partitionCount, workerCount);
         int myCount = workerToPartitionCount.get(this.hostName);
-        this.countAssignedLeases = myCount;
         int partitionsNeededForMe = target - myCount;
 
-        /*
-        Logger.InfoFormat(
-            "Host '{0}' {1} partitions, {2} hosts, {3} available leases, target = {4}, min = {5}, max = {6}, mine = {7}, will try to take {8} lease(s) for myself'.",
-            this.hostName,
-            partitionCount,
-            workerCount,
-            expiredLeases.Count,
-            target,
-            this.minScaleCount,
-            this.maxScaleCount,
-            myCount,
-            Math.Max(partitionsNeededForMe, 0));
-            */
+        if (expiredLeases.size() > 0) {
+            // We should try to pick at least one expired lease even if already overbooked when maximum partition count is not set.
+            // If other CFP instances are running, limit the number of expired leases to acquire to maximum 1 (non-greedy acquiring).
+            if (this.maxPartitionCount == 0 && (partitionsNeededForMe <= 0 || (partitionsNeededForMe > 1 && workerToPartitionCount.size() > 1))) {
```

Comment:
just to confirm my understanding is right here:

so if the customer ever set the maxScaleCount, then we will fail back to non-greedy algorithm
but if the customer has set the maxScaleCount, then we will going to use greedy algorithm?

Reply:
When maxScaleCount is set to a non-zero value, it's ok to be greedy because there's a cap to how much it can be assigned to one CFP instance. I agree that the system might still go though couple rounds of load balancing itself but much less than when no cap is set. The idea here is that as new instances are introduced into the system, they should start the work immediately on all the partitions that are available rather than only taking one at the time.